### PR TITLE
[ACR] Add OCI Artifact Source Context

### PR DESCRIPTION
**Pull request by akashsinghal**
**Description** 
 
Currently, ACR CLI does not support an [OCI artifact](https://github.com/opencontainers/artifacts) to be passed in as the source context or source location for any of the commands. We are adding support for this new third type of remote context and need to add CLI support for contexts provided of form `oci://myregistry.azurecr.io/artifact`.

**Testing Guide**  

`az acr build -f Dockerfile -r myregistry oci://myregistry.azurecr.io/artifact:v1` This command will take the provided source context (oci://) and use the Dockerfile downloaded from this source context to build. (Need to change registry with own registry. Need to change artifact link with own artifact uploaded to registry)
`az acr run -f acb.yaml -r myregistry oci://myregistry.azurecr.io/artifact:v2` This command will take the provided source context (oci://) and use the acb.yaml downloaded from this source context to run the task. (Need to change registry with own registry. Need to change artifact link with own artifact uploaded to registry)

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by akashsinghal** [comment]
@northtyphoon @shahzzzam

**Comment by northtyphoon** [comment]
@shahzzzam @jaysterp @rosanch

File: src/azure-cli/azure/cli/command_modules/acr/task.py
Comment (jaysterp) [inline src/azure-cli/azure/cli/command_modules/acr/task.py:85]: @akashsinghal this will throw a null exception if the above case is met and we set `context_path = None`. Please add a null check here before calling `.lower()`

**Comment by yonzhan** [comment]
ACR

**Comment by yungezz** [comment]
LGTM. Could you pls add some test?

**Comment by fengzhou-msft** [comment]
@akashsinghal can you fix the CI failure?

**Comment by akashsinghal** [comment]
@yungezz @fengzhou-msft Please do not merge this PR. These additions are reliant on new internal changes in ACR to be deployed to production and will be ready to merge after that has occurred. As for added tests, we will add a new test for this feature after global ACR deployment. Thanks! /cc: @northtyphoon @shahzzzam

**Comment by shahzzzam** [comment]
This Looks good to me now. It's ok to merge this. Internal changes to ACR have now been deployed.

cc: @northtyphoon @yungezz @fengzhou-msft

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls review and merge this PR? it's ready to merge now

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**
